src/utils/utils_part.h

```diff
@@ -61,7 +62,26 @@ static inline void ocf_part_sort(struct ocf_cache *cache)
 	ocf_lst_sort(&cache->lst_part);
 }
 
-static inline ocf_cache_mode_t ocf_part_get_cache_mode(struct ocf_cache *cache,
+static inline bool ocf_part_occ_check_limit(struct ocf_request *req)
+{
+	ocf_cache_t cache = req->cache;
+	uint32_t clines = ocf_engine_unmapped_count(req);
+	uint32_t max = req->cache->user_parts[req->part_id].config->max_size;
+	uint32_t curr_cached = env_atomic_read(
+				&cache->user_parts[req->part_id].runtime->valid_cnt);
+
+	uint32_t test = curr_cached + clines;
```

Comment:
This is hard to understand why this variable is named `test`. I'd make this entire check like this:
```c
uint32_t available_space = max ? max - curr_cached : 0;

if (clines > available_space) {
	...
```
BTW empty line between declarations and code is required.